nameresolution/sqlite/sqlite_migrations.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	sqlinternal "github.com/dapr/components-contrib/internal/component/sql"
+	sqlitemigrations "github.com/dapr/components-contrib/internal/component/sql/migrations/sqlite"
+	"github.com/dapr/kit/logger"
+)
+
+type migrationOptions struct {
+	HostsTableName    string
+	MetadataTableName string
+}
+
+// Perform the required migrations
+func performMigrations(ctx context.Context, db *sql.DB, logger logger.Logger, opts migrationOptions) error {
+	m := sqlitemigrations.Migrations{
+		Pool:              db,
+		Logger:            logger,
+		MetadataTableName: opts.MetadataTableName,
+		MetadataKey:       "nr-migrations",
+	}
+
+	return m.Perform(ctx, []sqlinternal.MigrationFn{
+		// Migration 0: create the hosts table
+		func(ctx context.Context) error {
+			logger.Infof("Creating hosts table '%s'", opts.HostsTableName)
+			_, err := m.GetConn().ExecContext(
+				ctx,
+				fmt.Sprintf(
+					`CREATE TABLE %[1]s (
+						registration_id TEXT NOT NULL PRIMARY KEY,
+						address TEXT NOT NULL,
+						app_id TEXT NOT NULL,
```

Comment:
We also need to include namespace

Reply:
Will do

Reply:
I just looked at the other components and it seems that no nameresolver besides the K8s supports the namespace. mDNS and Consul don't.

I think it should still be added to this component but it would require a lot of testing in the runtime.

Reply:
I just looked at the other components and it seems that no nameresolver besides the K8s supports the namespace. mDNS and Consul don't.

It doesn't seem to be possible to implement namespacing support at this stage. The runtime itself doesn't pass the namespace to the nameresolvers (none of them) when they are registered.

If namespacing support is required in these components, it should be done carefully and in a separate PR, as it is a significant change for the runtime too.

Reply:
I have created a separate issue to add namespacing support: https://github.com/dapr/components-contrib/issues/3179

It's not something we can do this in PR as it requires larger changes in the runtime too. 

However, in this PR I've added some "infra" already, including the constant that will then need to be used in the runtime.